AutoCollection/HeartBeat.ts

```diff
@@ -61,16 +73,88 @@ class HeartBeat {
             properties["appSrv_wsHost"] = process.env.WEBSITE_HOSTNAME || "";
         } else if (process.env.FUNCTIONS_WORKER_RUNTIME) { // Function apps
             properties["azfunction_appId"] = process.env.WEBSITE_HOSTNAME;
-        } 
-        this._client.trackMetric({name: Constants.HeartBeatMetricName, value: 0, properties: properties});
-        callback();
+        } else if (config) {
+            if (this._isVM === undefined) {
+                waiting = true;
+                this._getAzureComputeMetadata(config, () => {
+                    if (this._isVM && Object.keys(this._vmData).length > 0) { // VM
+                        properties["azInst_vmId"] = this._vmData["vmId"] || "";
+                        properties["azInst_subscriptionId"] = this._vmData["subscriptionId"] || "";
+                        properties["azInst_osType"] = this._vmData["osType"] || "";
+                        this._azInst_vmId = this._vmData["vmId"] || "";
+                        this._azInst_subscriptionId = this._vmData["subscriptionId"] || "";
+                        this._azInst_osType = this._vmData["osType"] || "";
+                    }
+                    this._client.trackMetric({name: Constants.HeartBeatMetricName, value: 0, properties: properties});
+                    callback();
+                });
+            } else if (this._isVM) {
+                properties["azInst_vmId"] = this._azInst_vmId;
```

Comment:
You need to continuously make requests to the azuremetadataservice because the vm fields can change if you find that the application in running in a VM. This is the whole point of tracking `_isVM`.

Reply:
oh ok.. i'm confused. @markwolff so should I call _getAzureComputeMetadata multiple times then?

Reply:
update in next commit